File: languages/div.lua
function div(x, y)
    if y == 0 then
        error("division by zero")
    end
    return x / y
end

function main()
    local x, y = 1.0, 0.0
    
    local status, result = pcall(div, x, y)
    
    if status then
        print("The result is: " .. result)
    else
        io.stderr:write("Division failed: " .. result .. "\n")
    end
end

main()

File: languages/div.swift
import Foundation

enum DivError: Error {
    case divisionByZero
    
    var localizedDescription: String {
        switch self {
        case .divisionByZero:
            return "division by zero"
        }
    }
}

func div(x: Double, y: Double) throws -> Double {
    if y == 0.0 {
        throw DivError.divisionByZero
    }
    return x / y
}

func main() {
    let x: Double = 1.0
    let y: Double = 0.0
    
    do {
        let z = try div(x: x, y: y)
        print("The result is: \(z)")
    } catch let error as DivError {
        fputs("Division failed: \(error.localizedDescription)\n", stderr)
    } catch {
        fputs("Unexpected error: \(error)\n", stderr)
    }
}

main()